lib/src/logger.dart

```diff
@@ -55,12 +55,11 @@ class Logger {
   /// with each log message.
   ///
   /// Additional [options] may be specified to configure the logger further.
-  factory Logger(LoggerCallback localCallback, [LoggerOptions options]) {
-    if (localCallback == null) throw ArgumentError.notNull('localCallback');
+  factory Logger(LoggerCallback localCallback, [LoggerOptions? options]) {
 
     // Wrap the callback to convert LogLevel
     void jsCallback(int level, String message, bool containsPii) {
-      localCallback(_getLogLevel(level), message, containsPii);
+      localCallback(_getLogLevel(level)!, message, containsPii);
```

Comment:
This is a tricky one. I'd rather Dart not throw a null error if `_getLogLevel` is missing a case. For example, if someone tried using a version of MSAL say 1 minor ahead that added a log level, it would throw an error if they used a slightly older version of this package.

I think it would be worth adding `LogLevel.unknown` for `_getLogLevel` to return instead of null.
The return in `_getLogLevel` could be easily be fixed (if `unknown` is the first value):
```diff
- return LogLevel.values[index];
+ return LogLevel.values[index + 1];
```

This would allow users to handle these cases on their own (such as treating unknown as info or whatever works for them). Functionally, this is the same as just making `level` in the callback nullable but is much more clear in my opinion (assuming `unknown` gets a doc comment explaining that it's not an MSAL value).

If you think using null instead of an unknown value is better, I'm not completely against it. But the `!` definitely needs to be removed.

Reply:
I'm not particularly comfortable in evaluating this decision, as my knowledge on MSAL or JS intertop can be brought to zero, so I'll add the `unknown` case with the appropriate documentation

still, I don't quite get how the user can access the log functionality if the version of MSAL he's using does not even expose the functionality? if he migrates from an older version of the library to the newer one (if it's not a patch that is), then he should expect that something might have changed, read the CHANGELOG and apply the changes necessary (just like with a LogLevel being added in the newer version he just upgraded to)

am I missing something here?

Reply:
My intention with this package is to allow users to use newer minor/patch versions of msal.js that just add/fix things. I agree that users should understand the changes they are getting into when using mismatched versions, but ultimately *additions* to msal.js should not break this package where possible.

Reply:
understood, thanks

does the updated solution fit?